src/core/engine/processors/mongo_to_postgresql_order_status_history.py

```diff
@@ -71,6 +73,23 @@ def load_data(self, transformed_data):
             mongo_uuids.append(
                 ObjectId(postgres_order_status_history.mongo_order_status_history_uuid)
             )
+
+            if postgres_order_status_history.from_status != OrderStatus.NEW_ORDER.name:
+                last_order_status_history = self.order_status_history_controller.get_last_status_history_by_order_id(
+                    postgres_order_status_history.order_id
+                )
+
+                last_order_status_history.to_status = (
+                    postgres_order_status_history.from_status
+                )
+                last_order_status_history.to_time = (
+                    postgres_order_status_history.from_time
+                )
+                self.order_status_history_controller.update_by_id(
```

Comment:
you should do a batch update instead

Reply:
done. To avoid the use of for loop i implemented a reduce function to take
the status of the new order status history to be assigned to the to status and to time of the last order status history using the order id.